src/fsharp/fsi/fsi.fs

```diff
@@ -821,7 +821,8 @@ type internal FsiDynamicCompiler
         errorLogger.AbortOnError();
             
         let fragName = textOfLid prefixPath 
-        let codegenResults = GenerateIlxCode (IlReflectBackend, isInteractiveItExpr, runningOnMono, tcConfig, topCustomAttrs, optimizedImpls, fragName, true, ilxGenerator)
+        let providedTypes, _staticLinker = Driver.StaticLinker.StaticLink(tcConfig, tcImports, ilGlobals)
```

Comment:
Essentially only the last parts of the static linking is done in fsi, see:

let mainmod3 = Morphs.morphILScopeRefsInILModuleMemoized ilGlobals
(NormalizeAssemblyRefs tcImports) ilxMainModule

these functions or similar are also done in the static linker, it's
slightly different for sci I guess because it's an incremental dynamic
assembly.


On Thursday, 21 January 2016, Don Syme <notifications@github.com> wrote:

> In src/fsharp/fsi/fsi.fs
> <https://github.com/Microsoft/visualfsharp/pull/882#discussion_r50476258>:
>
> > @@ -821,7 +821,8 @@ type internal FsiDynamicCompiler
> >          errorLogger.AbortOnError();
> >
> >          let fragName = textOfLid prefixPath
> > -        let codegenResults = GenerateIlxCode (IlReflectBackend, isInteractiveItExpr, runningOnMono, tcConfig, topCustomAttrs, optimizedImpls, fragName, true, ilxGenerator)
> > +        let providedTypes, _staticLinker = Driver.StaticLinker.StaticLink(tcConfig, tcImports, ilGlobals)
>
> Hmmm... I see "_staticLinker" is ignored here. I'm confused. I'm guessing
> that's because all the actual static linking is now done in IlxGen.fs.
>
> Now I'm wondering why the process of folding the provided types with the
> codegenResults isn't done in the StaticLink code? Doing it in IlxGen.fs
> seems somehow wrong - couldn't we have adjusted adjust implantTypeDef
> (which has been removed? because it's now done In IlxGen.fs?) to merge the
> type definitions where necessary?
>
> I guess I now don't understand why fsc.exe uses one static linking code,
> and fsi.exe uses another.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Microsoft/visualfsharp/pull/882/files#r50476258>.
>